libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/search.hpp

```diff
@@ -21,7 +23,306 @@
+    ///                     \endcode \n
+    ///                     The signature does not need to have const &, but
+    ///                     the function must not modify the objects passed to
+    ///                     it. The types \a Type1 and \a Type2 must be such
+    ///                     that objects of types \a FwdIter1 and \a FwdIter2 can
+    ///                     be dereferenced and then implicitly converted to
+    ///                     \a Type1 and \a Type2 respectively
+    /// \param proj1        Specifies the function (or function object) which
+    ///                     will be invoked for each of the elements of type
+    ///                     dereferenced \a FwdIter1 as a projection operation
+    ///                     before the actual predicate \a is invoked.
+    /// \param proj2        Specifies the function (or function object) which
+    ///                     will be invoked for each of the elements of type
+    ///                     dereferenced \a FwdIter2 as a projection operation
+    ///                     before the actual predicate \a is invoked.
+    ///
+    /// The comparison operations in the parallel \a search algorithm execute
+    /// in sequential order in the calling thread.
+    ///
+    /// \returns  The \a search algorithm returns a \a hpx::future<FwdIter> if the
+    ///           execution policy is of type \a task_execution_policy and
+    ///           returns \a FwdIter otherwise.
+    ///           The \a search algorithm returns an iterator to the beginning of
+    ///           the first subsequence [s_first, s_last) in range [first, last).
+    ///           If the length of the subsequence [s_first, s_last) is greater
+    ///           than the length of the range [first, last), \a last is returned.
+    ///           Additionally if the size of the subsequence is empty \a first is
+    ///           returned. If no subsequence is found, \a last is returned.
+    ///
+    template <typename FwdIter, typename Sent, typename FwdIter2,
+        typename Sent2, typename Pred = hpx::ranges::equal_to,
+        typename Proj1 = parallel::util::projection_identity,
+        typename Proj2 = parallel::util::projection_identity,
+        HPX_CONCEPT_REQUIRES_(
+            hpx::traits::is_forward_iterator<FwdIter>::value &&
+            hpx::traits::is_sentinel_for<Sent, FwdIter>::value &&
+            parallel::traits::is_projected<Proj1, FwdIter>::value &&
+            hpx::traits::is_forward_iterator<FwdIter2>::value &&
+            hpx::traits::is_sentinel_for<Sent2, FwdIter2>::value &&
+            parallel::traits::is_projected<Proj2, FwdIter2>::value &&
+            parallel::traits::is_indirect_callable<
+                hpx::execution::sequenced_policy,
+                Pred,
+                parallel::traits::projected<Proj1, FwdIter>,
+                parallel::traits::projected<Proj2, FwdIter2>
+            >::value
+        )>
```

Comment:
We've tended to leave this out from the doxygen declaration. The comment above instead documents the constraints.